pkg/config/config.go

```diff
@@ -37,38 +37,81 @@ var (
+	// IP address pool to use for pod IPs.
+	// This field is immutable after installation.
+	ClusterNetwork []ClusterNetworkEntry `json:"clusterNetwork"`
+
+	// IP address pool for services.
+	// Currently, we only support a single entry here.
+	// This field is immutable after installation.
+	ServiceNetwork []string `json:"serviceNetwork"`
+
+	// The port range allowed for Services of type NodePort.
+	// If not specified, the default of 30000-32767 will be used.
+	// Such Services without a NodePort specified will have one
+	// automatically allocated from this range.
+	// This parameter can be updated after the cluster is
+	// installed.
+	// +kubebuilder:validation:Pattern=`^([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])-([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$`
+	ServiceNodePortRange string `json:"serviceNodePortRange,omitempty"`
+
+	// Note: additional field to config v1 Network type
+	MTU string `json:"mtu"`
+}
 
-	AuditLogDir string `json:"auditLogDir"`
-	LogVLevel   int    `json:"logVLevel"`
+type ClusterNetworkEntry struct {
+	// The complete block for pod IPs.
+	CIDR string `json:"cidr"`
+}
 
-	Roles []string `json:"roles"`
+type DNS struct {
+	// baseDomain is the base domain of the cluster. All managed DNS records will
+	// be sub-domains of this base.
+	//
+	// For example, given the base domain `openshift.example.com`, an API server
+	// DNS record may be created for `cluster-api.openshift.example.com`.
+	//
+	// Once set, this field cannot be changed.
+	BaseDomain string `json:"baseDomain"`
+}
 
-	NodeName string `json:"nodeName"`
-	NodeIP   string `json:"nodeIP"`
+type APIServer struct {
+	// servingCert is the TLS cert info for serving secure traffic. If not specified, operator managed certificates
+	// will be used for serving secure traffic.
+	ServingCerts APIServerServingCerts `json:"servingCerts"`
+}
 
-	Cluster ClusterConfig `json:"cluster"`
+type APIServerServingCerts struct {
+	// namedCertificates references secrets containing the TLS cert info for serving secure traffic to specific hostnames.
+	// If no named certificates are provided, or no named certificates match the server name as understood by a client,
+	// the defaultServingCertificate will be used.
+	NamedCertificates []APIServerNamedServingCert `json:"namedCertificates,omitempty"`
+}
 
-	Manifests []string `json:"manifests"`
+type APIServerNamedServingCert struct {
+	// names is a optional list of explicit DNS names (leading wildcards allowed) that should use this certificate to
+	// serve secure traffic. If no names are provided, the implicit names will be extracted from the certificates.
+	// Exact names trump over wildcard names. Explicit names defined here trump over extracted implicit names.
+	Names []string `json:"names,omitempty"`
+}
 
-	Debug DebugConfig `json:"debug"`
+// Microshift specific config
+type MicroshiftConfig struct {
+	LogLevel int `json:"logLevel"`
```

Comment:
The current `LogLevel` is accepting an `int`. However, @dgrisonnet has a suggestion to model this LogLevel after [Operator v1 API](https://github.com/openshift/api/blob/622889ac07cf3dd8787679af0d3176bf292b3185/operator/v1/types.go#L55-L62) and then we will convert it the back to `int` verbosity. That way it will align better with OCP and users don't need why those numbers mean instead of using a plain English terminology such as "Normal", "Debug", "Trace", "TraceAll". What do you think? @dhellmann @deads2k @fzdarsky @benluddy

Reply:
In general I like plain English, but then it becomes unclear how these map to the Kubernetes [logging level conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md), or?

Reply:
Oh sorry. Forgot to include the convention code that is currently used in OCP: https://github.com/openshift/cluster-kube-apiserver-operator/blob/0378867a15c8f34b8286e1c92243a0a6bbdda383/pkg/operator/targetconfigcontroller/targetconfigcontroller.go#L492-L504

Reply:
Using an enum with conversion makes sense to me.